lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -892,7 +892,7 @@ SECOND_PASS:
                         else if (!HasNoMissingValues())
                         {
                             // Have we overwritten all the missing values?
-                            if (!ScanForMissingValues<T>(0, startOffset))
+                            if (!ScanForMissingValues<T>(0, startOffset) && !ScanForMissingValues<T>(startOffset + length, current->length))
```

Comment:
Why not ScanForMissingValues<T>(0, current->length)?

Reply:
Because indices from `startOffset -> startOffset + length - 1` would be overwritten by the memcopy/memset and we would have given up on memop if the source had missing values in the range being copied/set.

---
In reply to: [194876775](https://github.com/Microsoft/ChakraCore/pull/5300#discussion_r194876775) [](ancestors = 194876775)